operational/flexera/msp/customer_usage_insights_report/flexera_msp_customer_usage_insights_report.pt

```diff
@@ -0,0 +1,1425 @@
+  type "string"
+  category "Policy Settings"
+  label "Get Customer Org Processed Spend from MSP Parent Org"
+  description "Select whether to get customer org processed spend from the msp parent org or from customer orgs directly. Default is 'No' which gets costs from the parent org and customer orgs."
+  allowed_values "Yes", "No"
+  default "No"
+end
+
+###############################################################################
+# Authentication
+###############################################################################
+
+credentials "auth_flexera" do
+  schemes "oauth2"
+  label "Flexera"
+  description "Select Flexera One OAuth2 credentials"
+  tags "provider=flexera"
+end
+
+###############################################################################
+# Pagination
+###############################################################################
+
+pagination "pagination_flexera" do
+  get_page_marker do
+    body_path jmes_path(response, "nextPage")
+  end
+  set_page_marker do
+    uri true
+  end
+end
+
+###############################################################################
+# Datasources & Scripts
+###############################################################################
+
+# Get region-specific Flexera API endpoints
+datasource "ds_flexera_api_hosts" do
+  run_script $js_flexera_api_hosts, rs_optima_host
+end
+
+script "js_flexera_api_hosts", type: "javascript" do
+  parameters "rs_optima_host"
+  result "result"
+  code <<-EOS
+  host_table = {
+    "api.optima.flexeraeng.com": {
+      ui: "app.flexera.com",
+      flexera: "api.flexera.com",
+      fsm: "api.fsm.flexeraeng.com"
+    },
+    "api.optima-eu.flexeraeng.com": {
+      ui: "app.flexera.eu",
+      flexera: "api.flexera.eu",
+      fsm: "api.fsm-eu.flexeraeng.com"
+    },
+    "api.optima-apac.flexeraeng.com": {
+      ui: "app.flexera.au",
+      flexera: "api.flexera.au",
+      fsm: "api.fsm-apac.flexeraeng.com"
+    }
+  }
+
+  result = host_table[rs_optima_host]
+EOS
+end
+
+datasource "ds_dates" do
+  run_script $js_dates, $param_billing_period, $param_specific_period
+end
+
+script "js_dates", type: "javascript" do
+  parameters "param_billing_period", "param_specific_period"
+  result "result"
+  code <<-EOS
+  var now = new Date();
+
+  if (param_billing_period == "Specific Month") {
+    // If the user selected a specific month, we need to parse the date
+    var specific_month = new Date(param_specific_period);
+    // Set the start and end dates to the first day of the month
+    var start_at = new Date(specific_month.getFullYear(), specific_month.getMonth(), 1);
+    var end_at = new Date(specific_month.getFullYear(), specific_month.getMonth() + 1, 1);
+  } else if (param_billing_period == "Previous Month") {
+    // If the user selected previous month, we need to set the start and end dates to the first day of the previous month
+    var start_at = new Date(now.getFullYear(), now.getMonth() - 1, 1);
```

Comment:
using getMonth() and then adding/substracting from it is not recommended. Pain in the butt, I know, but it causes issues where it sometimes jumps more than a full month due to the weirdness of how it handles certain days.